weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -1706,13 +1745,22 @@ def _transform_external_field_to_internal_field(
     return field, param_builder, raw_fields_used
 
 
+class FilterToConditions(BaseModel):
+    having_conditions: list[str]
+    start_event_conditions: list[str]
+    end_event_conditions: list[str]
+    fields_used: set[str]
+
+
 def _process_calls_filter_to_conditions(
     filter: tsi._CallsFilter,
     param_builder: typing.Optional[ParamBuilder] = None,
-) -> tuple[list[str], set[str]]:
+) -> FilterToConditions:
```

Comment:
This method has been refactored to return specific conditions where possible, using correct namespacing